openedx/core/djangoapps/persistenttask/models.py

```diff
@@ -0,0 +1,32 @@
+"""
+Models to support persistent tasks.
+"""
+
+from django.db import models
+from jsonfield import JSONField
+
+
+class FailedTask(models.Model):
+    """
+    Representation of tasks that have failed
+    """
+    task_name = models.CharField(max_length=255)
+    task_id = models.CharField(max_length=255, db_index=True)
+    args = JSONField(blank=True)
+    kwargs = JSONField(blank=True)
+    exc = models.CharField(max_length=255)
+    datetime_failed = models.DateTimeField(db_index=True)
+    datetime_resolved = models.DateTimeField(blank=True, null=True, default=None)
+
+    class Meta(object):
+        index_together = [
+            (u'task_name', u'exc'),
```

Comment:
I don't have any data on it.  The value will be the repr of the exception, which will contain task specific stuff.  Since it's the last column in the index, we can still leverage the index with a prefix of it, with something along the lines of:

    SELECT * from celery_utils_failedtask WHERE task_name = 'my_task' AND exc LIKE 'ValueError(\'No result found for%';

But the actual indexed value will be different for each row.  Does that impact the size of the index?  I thought MySQL stored the body of the indexed value for each row regardless of whether they are unique, but I admit I'm not sure about that.

Since part of the plan for this is to delete rows that have been resolved for 30 days, I don't think we need to worry about unbounded table size.

If I set up a sandbox, do we have a way to load useful sample course data, so I can inject a code error, visit a few progress pages, and get lots of entries in it?